http-api/src/lib.rs

```diff
@@ -220,13 +321,115 @@ async fn peer_handler(peer_id: PeerId) -> Result<impl warp::Reply, warp::Rejecti
     Ok(warp::reply::json(&response))
 }
 
+/// `GET /v1/sessions`
+async fn session_state_handler(
+    ctx: Context,
+    id: String,
+) -> Result<impl warp::Reply, warp::Rejection> {
+    let sessions = ctx.sessions.write().await;
+    let session = sessions
+        .get(&id)
+        .ok_or_else(|| warp::reject::custom(Error::Siwe("Session not found")))?;
+
+    if let AuthState::Authorized(message) = session {
+        Ok(warp::reply::json(
+            &json!({ "id": id.clone(), "session": message }),
+        ))
+    } else {
+        Err(warp::reject::custom(Error::Siwe("Not authorized")))
+    }
+}
+
+/// `POST /v1/sessions`
+async fn session_creation_handler(ctx: Context) -> Result<impl warp::Reply, warp::Rejection> {
+    let expiration_time = ctx.now()? + UNAUTHORIZED_SESSIONS_EXPIRATION;
+    let mut sessions = ctx.sessions.write().await;
+    let (session_id, nonce) = ctx
+        .create_session(&mut sessions, expiration_time)
+        .map_err(|_| warp::reject::custom(Error::Siwe("Not able to create session")))?;
+
+    let response = json!({ "id": session_id, "nonce": nonce });
+
+    Ok(warp::reply::json(&response))
+}
+
+/// `PUT /v1/sessions/:session_id`
+async fn session_signin_handler(
+    ctx: Context,
+    id: String,
+    request: AuthRequest,
+) -> Result<impl warp::Reply, warp::Rejection> {
+    // Get unauthenticated session data, return early if not found
+    let mut sessions = ctx.sessions.write().await;
+    let session = sessions
+        .get(&id)
+        .ok_or_else(|| warp::reject::custom(Error::Siwe("Session not found")))?;
+
+    if let AuthState::Unauthorized(auth) = session {
+        let message = Message::from_str(request.message.as_str())
+            .map_err(|_| Error::Siwe("Not able to parse Siwe message"))?;
+        // TODO: Not sure if using an env variable is the right thing to do.
+        let host =
+            env::var("RADICLE_DOMAIN").map_err(|_| Error::Siwe("No host env variable found"))?;
```

Comment:
This is probably fine for now, but I'd improve the error as `RADICLE_DOMAIN is not set`